host_provider/providers/azure.py

```diff
@@ -0,0 +1,384 @@
+from collections import namedtuple
+from requests.status_codes import codes as response_created
+import os
+import json
+import re
+from collections import OrderedDict
+from contextlib import suppress
+from host_provider.credentials.azure import CredentialAddAzure, CredentialAzure
+from host_provider.common.azure import AzureConnection
+from host_provider.models import Host
+from host_provider.providers import ProviderBase
+from host_provider.settings import HOST_ORIGIN_TAG
+from libcloud.compute.types import Provider
+from host_provider.clients.team import TeamClient
+from pathlib import Path
+
+
+class NodeFoundError(Exception):
+    pass
+
+
+class OperationNotAllowed(Exception):
+    pass
+
+
+class DeployVmError(Exception):
+    pass
+
+
+class OfferingNotFoundError(Exception):
+    pass
+
+
+class InvalidParameterError(Exception):
+    pass
+
+
+class JsonTemplates(object):
+    def __init__(self, path="host_provider/templates/azure/version/"):
+        self.path = path
+
+    def list_files(self, version):
+        files_version = os.path.join(self.path, f"{version}/")
+        files = [path for path in Path(files_version).rglob("*.json")]
+        return files
+
+    def load_json(self, json_file):
+        data = None
+
+        with open(json_file) as fp:
+            data = json.load(fp)
+        return data
+
+    def dumps_json(self, obj):
+        if isinstance(obj, dict):
+            return json.dumps(obj)
+        return None
+
+
+class AzureProvider(ProviderBase):
+    BasicInfo = namedtuple("AzureBasicInfo", "id")
+    azClient = None
+    connCls = AzureConnection
+    paths = {
+        "action_getnode": "subscriptions/{}/providers/Microsoft.Compute/virtualMachines?api-version={}",
+        "imageid_parseimage": "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Compute/galleries/{}/images/{}/versions/{}",
+        "networkid_parseimage":"/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/networkInterfaces/{}",
+        "action_offeringto":"subscriptions/{}/providers/Microsoft.Compute/locations/{}/vmSizes?api-version={}",
+        "id_parsenic":"/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets/{}",
+        "action_createnic":"subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/networkInterfaces/{}?api-version={}",
+        "action_getnetwork":"subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}?api-version={}",
+        "action_listvm":"subscriptions/{}/providers/Microsoft.Compute/virtualMachines/?api-version={}&statusOnly={}",
+        "action_deployvm":"subscriptions/{}/resourceGroups/{}/providers/Microsoft.Compute/virtualMachines/{}?api-version={}"
```

Comment:
I think it is better to put this variable in the common/azure.py file and import it here. Just to avoid polluting this file that has business rules.